lavalink/PlayerManager.py

```diff
@@ -114,10 +115,10 @@ def add_at(self, index: int, requester: int, track: dict):
             await self.stop()
             await self._lavalink.dispatch_event(QueueEndEvent(self))
         else:
-            if self.shuffle:
+            if self.shuffle and not ignore_shuffle:
                 track = self.queue.pop(randrange(len(self.queue)))
             else:
-                track = self.queue.pop(0)
+                track = self.queue.pop(min(track_index, len(self.queue) - 1))

```

Comment:
Is this supposed to have the 2nd parameter?

Reply:
No? Did I miss something?